gears/src/baseapp/mod.rs

```diff
@@ -42,6 +47,7 @@ pub struct BaseApp<
 > {
     multi_store: Arc<RwLock<MultiStore<RocksDB, SK>>>,
     height: Arc<RwLock<u64>>,
+    block_gas_meter: Arc<RwLock<Box<dyn GasMeter>>>,
```

Comment:
Does check_tx need a block_gas_meter? If it does it can't use the same one as deliver_tx.